config/commands.js

```diff
@@ -676,18 +676,21 @@ var commands = exports.commands = {
 					break;
 
 				case 'moves':
+					var move = {};
+					for (var i in searches[search]) {
+						move[i] = Tools.getMove(i);
+						if (!move[i].exists) return this.sendReplyBox("'" + move[i] + "' is not a known move.");
+					}
 					for (var mon in dex) {
 						var template = Tools.getTemplate(dex[mon].id);
 						if (!template.learnset) template = Tools.getTemplate(template.baseSpecies);
 						if (!template.learnset) continue;
 						for (var i in searches[search]) {
-							var move = Tools.getMove(i);
```

Comment:
why not keep the new cache object as `moves`, and this line as `var move = moves[i];`
So, you need not change the further lines from `move` to `move[i]`.